go.mod

```diff
@@ -3,16 +3,14 @@ module go.temporal.io/api
 go 1.16
 
 require (
-	github.com/gogo/googleapis v1.4.1
-	github.com/gogo/protobuf v1.3.2
-	github.com/gogo/status v1.1.1
 	github.com/golang/mock v1.6.0
-	github.com/golang/protobuf v1.5.3
-	github.com/grpc-ecosystem/grpc-gateway v1.16.0
+	github.com/google/go-cmp v0.5.9
```

Comment:
Is this only for tests? I know it may seem a bit much, but this is a really core library to a lot of people's projects, so ideally we can not add any dependencies we don't have to (even if it's a transitive dependency of something we already have, that's better than explicit)

Reply:
It's used for tests because you cannot easily compare protobuf objects; testify's Equal.* method don't work as they use reflection and so will compare the cached serialization state of the protobuf objects (even though its not exported).

I can write a simple reflection-based comparator if we need, but I'm introducing cmp in our downstream repos for the same reason

Reply:
Done.